src/udev/udev-builtin-net_id.c

```diff
@@ -243,9 +243,29 @@ static bool is_pci_ari_enabled(sd_device *dev) {
         return streq(a, "1");
 }
 
+static bool is_pci_bridge(sd_device *dev) {
+        const char *v, *p;
+
+        if (sd_device_get_sysattr_value(dev, "modalias", &v) < 0)
+                return false;
+
+        if (!startswith(v, "pci:"))
+                return false;
+
+        p = strrchr(v, 's');
+        if (!p)
+                return false;
+        if (*++p != 'c')
+                return false;
+
+        /* PCI device subclass 04 corresponds to PCI bridge */
+        return strneq(++p, "04", 2);
+}
+
 static int dev_pci_slot(sd_device *dev, struct netnames *names) {
         unsigned long dev_port = 0;
-        unsigned domain, bus, slot, func, hotplug_slot = 0;
+        unsigned domain, bus, slot, func;
+        int hotplug_slot = -1;
```

Comment:
there's some type confusion around this. (was there previously too). In some cases we assume its signed, in others unsigned. When you touch this, please clean this up. i.e. if this is signed now then please adjust "unsigned i" in the loop to be "int i", too, and use safe_atoi_full() rather than safe_atou_full() for parsing it, too.